### PR TITLE
Use tag instead of name in Location

### PR DESCRIPTION
**Pull request by rsheeter**
Name matches how some sources do it but Tag feels much more natural so translate to tag more aggressively.

IMHO #347 makes more sense after this. I think this is a good change regardless of whether we proceed with #347.

JMM.

### CODE REVIEW

File: fontbe/src/orchestration.rs [hunk 1]
```diff
@@ -705,7 +708,7 @@ mod tests {
     fn non_default_region() -> VariationRegion {
         let mut region = VariationRegion::default();
         region.insert(
-            "Weight".to_string(),
+            Tag::from_str("wght").unwrap(),
```
Comment (cmyr) [inline fontbe/src/orchestration.rs:711]: I'd prefer writing this as,


```suggestion
            Tag::new(b"wght"),
```

File: glyphs2fontir/src/source.rs [hunk 2]
```diff
@@ -996,8 +993,8 @@ mod tests {
             (72.0, 700.0),
         ] {
             let mut loc = UserLocation::new();
-            loc.set_pos("Optical Size", UserCoord::new(*opsz));
-            loc.set_pos("Weight", UserCoord::new(*wght));
+            loc.set_pos(Tag::from_str("opsz").unwrap(), UserCoord::new(*opsz));
```
Comment (cmyr) [inline glyphs2fontir/src/source.rs:996]: ditto these could just use`Tag::new`, which is infallible.

File: ufo2fontir/src/source.rs [hunk 3]
```diff
@@ -1252,11 +1258,12 @@ mod tests {
     fn add_design_location(
         add_to: &mut HashMap<PathBuf, Vec<DesignLocation>>,
         glif_file: &str,
-        axis: &str,
+        tag: &str,
```
Comment (cmyr) [inline ufo2fontir/src/source.rs:1261]: would it make sense to ask the caller to pass in a `Tag`?

**Review by cmyr — APPROVED** [review]
I think this makes sense, tags are used much more consistently than names, and means we don't need to worry about borrowing/cloning all over the place.

In this PR you're using `Tag::from_str` as a general constructor, but that method is really only intended for use when parsing; if you have a known-good input (a literal) you should prefer `Tag::new` which takes a `&[u8; 4]` and can't fail.

File: glyphs2fontir/src/source.rs (hunk 2 above)
Comment (rsheeter) [reply glyphs2fontir/src/source.rs:996]: Ty, will fix.

File: ufo2fontir/src/source.rs (hunk 3 above)
Comment (rsheeter) [reply ufo2fontir/src/source.rs:1261]: Yes, will fix.

**Comment by rsheeter** [comment]
Force push to rebase after merging #346

**Pull request merged**